InterruptingEventSubprocessTest#shouldCloseEventSubscriptions[message event subprocess] is flaky

**Summary**

- How often does the test fail? so far 2 times in a row
- Does it block your work? not yet
- Do we suspect that it is a real failure? can't say, probably not?

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
java.lang.AssertionError: 
Expecting ArrayList:
  [CREATE, CREATED]
to contain:
  [DELETED]
but could not find the following element(s):
  [DELETED]
	at io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest.shouldCloseEventSubscriptions(InterruptingEventSubprocessTest.java:354)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:43)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:82)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:73)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:199)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:193)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
</pre>
</details>

**Hypotheses**

No idea

**Logs**

<details><summary>Logs</summary>
<pre>
20:04:57.179 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@4f6695e5)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@4f398dce, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@3960554e, configuration: Configuration(false)]
20:04:57.474 [] INFO  io.zeebe.test - Test failed, following records were exported:
20:04:57.767 [] INFO  io.zeebe.test - Compact log representation:
--------
	['C'ommand/'E'event/'R'ejection] [valueType] [intent] - #[position]->#[source record position]  P[partitionId]K[key] - [summary of value]
	P9K999 - key; #999 - record position; "ID" element/process id; @"elementid"/[P9K999] - element with ID and key
	Keys are decomposed into partition id and per partition key (e.g. 2251799813685253 -> P1K005). If single partition, the partition is omitted.
	Long IDs are shortened (e.g. 'startEvent_5d56488e-0570-416c-ba2d-36d2a3acea78' -> 'star..acea78'
--------
C DEPLOYMENT             CREATE            - #139->  -1   -1 - process.xml
E PROC                   CREATED           - #140->#139 K051 - process.xml->"proc" (version:4)
E DEPLOYMENT             CREATED           - #141->#139 K052 - process.xml
E DEPLOYMENT             FULLY_DISTR       - #142->#139 K052 - 
C PROC_INSTANCE_CREATION CREATE            - #143->  -1   -1 - new <process "proc"> with variables: {key=123}
E VARIABLE               CREATED           - #144->#143 K054 - key->123 in <process [K053]>
E PROC_INSTANCE          ELMNT_ACTIVATING  - #145->#143 K053 - PROCESS "proc" in <process "proc"[K053]>
E PROC_INSTANCE_CREATION CREATED           - #146->#143 K055 - new <process "proc"> with variables: {key=123}
C TIMER                  CREATE            - #147->#145   -1 -  @"event_sub_start"[K053]  in <process "K051"[K053]> due T20:08:57.274
E PROC_MSG_SUB           CREATING          - #148->#145   -1 - "other-message" (inter.) correlationKey: 123 @[K053] in <process "proc"[K053]> (no vars)
C TIMER                  CREATE            - #149->#145   -1 -  @"other-timer"[K053]  in <process "K051"[K053]> due 2021-03-29T20:07:57.274
E PROC_INSTANCE          ELMNT_ACTIVATED   - #150->#145 K053 - PROCESS "proc" in <process "proc"[K053]>
C MSG_SUB                CREATE            - #151->  -1   -1 - "other-message" (inter.) correlationKey: 123 @[K053] in <process "proc"[K053]> (no vars)
E TIMER                  CREATED           - #152->#147 K056 -  @"event_sub_start"[K053]  in <process "K051"[K053]> due T20:08:57.274
E TIMER                  CREATED           - #153->#149 K057 -  @"other-timer"[K053]  in <process "K051"[K053]> due 2021-03-29T20:07:57.274
E PROC_INSTANCE          ELMNT_ACTIVATING  - #154->#150 K058 - START_EVENT "start_proc" in <process "proc"[K053]>
E MSG_SUB                CREATED           - #155->#151 K059 - "other-message" (inter.) correlationKey: 123 @[K053] in <process "proc"[K053]> (no vars)
C PROC_MSG_SUB           CREATE            - #156->  -1   -1 - "other-message" (inter.) @[K053] in <process ?> (no vars)
E PROC_INSTANCE          ELMNT_ACTIVATED   - #157->#154 K058 - START_EVENT "start_proc" in <process "proc"[K053]>
E PROC_MSG_SUB           CREATED           - #158->#156   -1 - "other-message" (inter.) @[K053] in <process ?> (no vars)
E PROC_INSTANCE          ELMNT_COMPLETING  - #159->#157 K058 - START_EVENT "start_proc" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_COMPLETED   - #160->#159 K058 - START_EVENT "start_proc" in <process "proc"[K053]>
E PROC_INSTANCE          SEQ_FLOW_TAKEN    - #161->#160 K060 - SEQUENCE_FLOW "sequenc..5e3ab6e" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_ACTIVATING  - #162->#161 K061 - SERVICE_TASK "task" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_ACTIVATED   - #163->#162 K061 - SERVICE_TASK "task" in <process "proc"[K053]>
C JOB                    CREATE            - #164->#163   -1 -  "type" @"task"[K061] 3 retries, in <process "proc"[K053]> (no vars)
E JOB                    CREATED           - #165->#164 K062 - K062 "type" @"task"[K061] 3 retries, in <process "proc"[K053]> (no vars)
C TIMER                  TRIGGER           - #166->#164 K056 -  @"event_sub_start"[K053]  in <process "K051"[K053]> due T20:08:57.274
E TIMER                  TRIGGERED         - #167->#166 K056 -  @"event_sub_start"[K053]  in <process "K051"[K053]> due T20:08:57.274
E PROC_INSTANCE          EVENT_OCCURRED    - #168->#166 K063 - START_EVENT "event_sub_start" in <process "proc"[K053]>
C TIMER                  CANCEL            - #169->#168 K057 -  @"other-timer"[K053]  in <process "K051"[K053]> due 2021-03-29T20:07:57.274
E PROC_INSTANCE          ELMNT_TERMINATING - #170->#168 K061 - SERVICE_TASK "task" in <process "proc"[K053]>
E TIMER                  CANCELED          - #171->#169 K057 -  @"other-timer"[K053]  in <process "K051"[K053]> due 2021-03-29T20:07:57.274
C JOB                    CANCEL            - #172->#170 K062 - K062 "type" @"task"[K061] 3 retries, in <process "proc"[K053]> (no vars)
E PROC_INSTANCE          ELMNT_TERMINATED  - #173->#170 K061 - SERVICE_TASK "task" in <process "proc"[K053]>
E JOB                    CANCELED          - #174->#172 K062 - K062 "type" @"task"[K061] 3 retries, in <process "proc"[K053]> (no vars)
E PROC_INSTANCE          ELMNT_ACTIVATING  - #175->#173 K064 - SUB_PROCESS "event_sub_proc" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_ACTIVATED   - #176->#175 K064 - SUB_PROCESS "event_sub_proc" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_ACTIVATING  - #177->#176 K065 - START_EVENT "event_sub_start" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_ACTIVATED   - #178->#177 K065 - START_EVENT "event_sub_start" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_COMPLETING  - #179->#178 K065 - START_EVENT "event_sub_start" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_COMPLETED   - #180->#179 K065 - START_EVENT "event_sub_start" in <process "proc"[K053]>
E PROC_INSTANCE          SEQ_FLOW_TAKEN    - #181->#180 K066 - SEQUENCE_FLOW "sequenc..0ffd077" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_ACTIVATING  - #182->#181 K067 - END_EVENT "event_sub_end" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_ACTIVATED   - #183->#182 K067 - END_EVENT "event_sub_end" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_COMPLETING  - #184->#183 K067 - END_EVENT "event_sub_end" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_COMPLETED   - #185->#184 K067 - END_EVENT "event_sub_end" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_COMPLETING  - #186->#185 K064 - SUB_PROCESS "event_sub_proc" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_COMPLETED   - #187->#186 K064 - SUB_PROCESS "event_sub_proc" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_COMPLETING  - #188->#187 K053 - PROCESS "proc" in <process "proc"[K053]>
E PROC_INSTANCE          ELMNT_COMPLETED   - #189->#188 K053 - PROCESS "proc" in <process "proc"[K053]>
C MSG_SUB                DELETE            - #190->  -1   -1 - "other-message" (inter.) @[K053] in <process ?> (no vars)
C MSG_SUB                DELETE            - #191->  -1   -1 - "other-message" (inter.) @[K053] in <process ?> (no vars)
E MSG_SUB                DELETED           - #192->#190 K059 - "other-message" (inter.) correlationKey: 123 @[K053] in <process "proc"[K053]> (no vars)
NOT_FOUND Expected to close message subscription for element with key '2251799813685301' and message name 'other-message', but no such message subscription exists#193->#191   -1 - 
C PROC_MSG_SUB           DELETE            - #194->  -1   -1 - "other-message" (inter.) @[K053] in <process ?> (no vars)
C PROC_MSG_SUB           DELETE            - #195->  -1   -1 - "other-message" (inter.) @[K053] in <process ?> (no vars)
E PROC_MSG_SUB           DELETED           - #196->#194   -1 - "other-message" (inter.) @[K053] in <process ?> (no vars)
NOT_FOUND Expected to delete process message subscription for element with key '2251799813685301' and message name 'other-message', but no such subscription was found.#197->#195   -1 - 
--------
Decomposed keys (for debugging):
  -1 <-> -1
K051 <-> 2251799813685299
K052 <-> 2251799813685300
K053 <-> 2251799813685301
K054 <-> 2251799813685302
K055 <-> 2251799813685303
K056 <-> 2251799813685304
K057 <-> 2251799813685305
K058 <-> 2251799813685306
K059 <-> 2251799813685307
K060 <-> 2251799813685308
K061 <-> 2251799813685309
K062 <-> 2251799813685310
K063 <-> 2251799813685311
K064 <-> 2251799813685312
K065 <-> 2251799813685313
K066 <-> 2251799813685314
K067 <-> 2251799813685315
</pre>
</details>

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessTest.java
@@ -377,22 +377,20 @@ public class InterruptingEventSubprocessTest {
 
     // then
     assertThat(
-            RecordingExporter.records()
-                .limitToProcessInstance(processInstanceKey)
-                .messageSubscriptionRecords()
+            RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.DELETED)
                 .withProcessInstanceKey(processInstanceKey)
-                .withMessageName("other-message"))
-        .extracting(Record::getIntent)
-        .contains(MessageSubscriptionIntent.DELETED);
+                .withMessageName("other-message")
+                .findFirst())
+        .describedAs("Expected the message subscription to be deleted")
+        .isPresent();
 
     assertThat(
-            RecordingExporter.records()
-                .limitToProcessInstance(processInstanceKey)
-                .timerRecords()
+            RecordingExporter.timerRecords(TimerIntent.CANCELED)
                 .withProcessInstanceKey(processInstanceKey)
-                .withHandlerNodeId("other-timer"))
-        .extracting(Record::getIntent)
-        .contains(TimerIntent.CANCELED);
+                .withHandlerNodeId("other-timer")
+                .findFirst())
+        .describedAs("Expected the timer to be canceled")
+        .isPresent();
   }
 
   private static void assertEventSubprocessLifecycle(final long processInstanceKey) {

